xray/backends/scipy_.py

```diff
@@ -26,6 +27,29 @@ def _decode_attrs(d):
                        for (k, v) in iteritems(d))
 
 
+class ScipyArrayWrapper(NumpyArrayAdapter):
+    def __init__(self, netcdf_file, variable_name):
+        self.netcdf_file = netcdf_file
+        self.variable_name = variable_name
+
+    @property
+    def array(self):
+        # We can't store the actual netcdf_variable object or its data array,
+        # because otherwise scipy complains about variables or files still
+        # referencing mmapped arrays when we try to close datasets without
+        # having read all data in the file.
+        return self.netcdf_file.variables[self.variable_name].data
+
+    def __getitem__(self, key):
+        data = super(ScipyArrayWrapper, self).__getitem__(key)
+        if self.netcdf_file.use_mmap:
```

Comment:
This works pretty cleanly, but in the process I discovered that there's no reliable way to check whether a numpy array is mem-mapped or not. The arrays scipy.io.netcdf returns are not `np.memmap` instances, but their data does point to a file on disk.